### ISSUE TITLE
[BUG] Dashy not reading valid config or saving config properly

### ISSUE THREAD

**Issue by stancubed** [Lissy93/dashy#831]
### Environment

Self-Hosted (Docker)

### System

Ubuntu 20.04.4 LTS

### Version

2.1.1

### Describe the problem

Dashy is not reading my valid config.yml file when loading, so it shows "No Data Configured" and using the UI to view the config, it shows only:
`[
    "object Object"
]`

When manually pasting in the valid config object to the "Edit Config" tool, it validates in UI but saving it only saves as:
`[
    "object Object"
]`

There is no inode issue as both inside/outside the container the inode is the same. There are no errors on build or validation, none in the console, and none during manual build. Pasting the build log below!

[build_log.txt](https://github.com/Lissy93/dashy/files/9255018/build_log.txt)

Happy to test anything or check somewhere else for more logs!

### Please tick the boxes

- [X] You have explained the issue clearly, and included all relevant info
- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy
- [X] You've checked that this [issue hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide ![important](https://img.shields.io/badge/Important!-F6094E)
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by stancubed** [comment]
I should mention that the build log is from a recent test of 2.1.0 but this is happening on both 2.1.1 and 2.1.0.

**Comment by Lissy93** [comment]
#799

**Comment by stancubed** [comment]
The strange thing is that I had this working for a few days until today! What could have caused this if it wasn't a code change?

**Comment by Lissy93** [comment]
Yeah, this is really weird. I've not come across this issue before. The ` [ "object Object" ]` means that the JSON hasn't been parsed/ not converted to string.

Everything in that log file looks alright, I guess it's a client-side bug. Could you share the contents of your browser console? ([here's how](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#how-to-open-browser-console)).

There are a lot of problems with the current config system, the main one being that it's really slow to apply changed especially on low spec devices. I'm about 50% through re-writing it all at the moment, and that _should_ fix everything. But I understand it's frustrating in the meantime.

**Comment by stancubed** [comment]
Okay that's sort of what I was thinking too, but it's weird that there aren't any logged issues during the save process either. Attaching the logs here!

It could totally be a client issue -- is there where to config is translated before saving?

[dashy-1659632635826.log](https://github.com/Lissy93/dashy/files/9261778/dashy-1659632635826.log)

**Comment by v1per404** [comment]
It's been a couple of months since the issue exist. I think it has something to do with the docker image, maybe try to make the config file in a separate folder so we can bind the folder instead a file, like other docker images that I have used?

**Comment by desrod** [comment]
Chiming in as well, just installed Dashy here using the default instructions and it does not allow me to edit/save/load or change the configuration as well. Changes to `~/dashy-config.yml` on the host side are seen inside the editor, but never reflected in the UI, and it always reports "No Data Configured". 

I came here because I initially tried using "Homepage", but that's an even worse dumpster-fire of an attempt at service aggregation.

**Comment by CrazyWolf13** [comment]
Hi,
Please check out Dashy V3, it compledtely changes how config is fetched and saved, there are no more rebuilds required.

Keep in mind V3 has breaking changes:
https://github.com/Lissy93/dashy/releases/tag/3.0.0

If this is still an issue with V3.0.1 comment here and I'll reopen this.

**Comment by TheRealGramdalf** [comment]
This is an issue I'm running up against in v3.1.1. I'm using pure JSON in the `conf.yml`, which is supported by the YAML spec (you can even use embedded JSON inside a typical YAML document), but there seem to be some issues with the client parsing - it passes the config validator perfectly, but fails with the same symptoms as above (no data configured, config edit screen shows `object Object`) when built.

I should mention that I'm also running this statically via NGINX, rather than yarn directly. I'm not _expecting_ anything to change (though if I can use pure JSON that would be nice), I just thought I'd mention it in case it points to an issue somewhere. I can provide plenty of details upon request if needed.

**Comment by secretcitydev** [comment]
Also seeing this issue in v3.1.1 - I've replicated across multiple hosts. Config.yml file is valid, and exists within the docker container at the expected location, however only the default getting started config is presented on the dashboard. 

docker logs dashy
yarn run v1.22.19
$ NODE_OPTIONS=--openssl-legacy-provider npm-run-all --parallel build start
$ node server
$ NODE_OPTIONS=--openssl-legacy-provider vue-cli-service build

Checking config file against schema...
✔️ Config file is valid, no issues found

SSL Not Enabled: Public key not present


 ██████╗  █████╗ ███████╗██╗  ██╗██╗   ██╗
 ██╔══██╗██╔══██╗██╔════╝██║  ██║╚██╗ ██╔╝
 ██║  ██║███████║███████╗███████║ ╚████╔╝
 ██║  ██║██╔══██║╚════██║██╔══██║  ╚██╔╝
 ██████╔╝██║  ██║███████║██║  ██║   ██║
 ╚═════╝ ╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝   ╚═╝

*******************************************************************************************
Welcome to Dashy! 🚀
Your new dashboard is now up and running with Docker
*******************************************************************************************


Using Dashy V-3.1.1. Update Check Complete
✅ Dashy is Up-to-Date
